src/ui/features/xp-drop/XpDropClaimFlow/XpDropClaim.tsx

```diff
@@ -0,0 +1,510 @@
+        initiator: INTERNAL_ORIGIN,
+        clientScope: null,
+      });
+      await ZerionAPI.claimRetro({
+        address: selectedWallet.address,
+        signature,
+      });
+      // Typically, after we call `/wallet/claim-retro/v1` it takes asecond or two
+      // before `membership.retro` becomes `null` (in response to `/wallet/get-meta/v1`).
+      // To account for this possible delay, we're adding a short wait here.
+      // Otherwise, we might still see the wallet for which we just claimed the XP drop in
+      // the wallet selector when we click on the "Next Wallet" button.
+      await wait(2000);
+    },
+    onSuccess: async () => {
+      // Refetch the wallet metadata to update the list of wallets eligible for the XP drop.
+      await walletsMetaQuery.refetch();
+      // If this is the last wallet, we want to display the success view.
+      if (eligibleAddresses.length <= 1) {
+        navigate('/xp-drop/claim/success');
+      }
+    },
+  });
+
+  const selectedWalletMeta = useMemo(
+    () =>
+      selectedWallet
+        ? walletsMeta?.find(
+            (meta) =>
+              normalizeAddress(meta.address) ===
+              normalizeAddress(selectedWallet.address)
+          )
+        : null,
+    [selectedWallet, walletsMeta]
+  );
+
+  const membership = selectedWalletMeta?.membership;
+  const [retro, setRetro] = useState(selectedWalletMeta?.membership.retro);
+
+  useEffect(() => {
+    // We should never set the current retrodrop info to null or undefined,
+    // so that we can always display the success state.
+    if (membership?.retro) {
+      setRetro(membership.retro);
+    }
+  }, [membership, retro]);
+
+  if (
+    walletGroupsQuery.isLoading ||
+    !selectedWallet ||
+    !walletsMeta ||
+    !eligibleWalletGroups ||
+    !eligibleAddresses ||
+    !membership
+  ) {
+    return null;
+  }
+
+  if (!retro) {
+    return (
+      <>
+        <NavigationTitle title="Not eligible" backTo="/overview" />
+        <FillView>
+          <UIText
+            kind="body/regular"
+            color="var(--neutral-500)"
+            style={{ textAlign: 'center' }}
+          >
+            The selected wallet is not eligible for an XP drop. <br />
+            Please select another wallet and try again.
+          </UIText>
+        </FillView>
+      </>
+    );
+  }
+
+  return (
+    <>
+      <PageColumn>
+        <NavigationTitle
+          title={
+            personalSignMutation.isSuccess
+              ? 'Congratulations!'
+              : 'Claim Your XP'
+          }
+          backTo="/overview"
+        />
+        <PageTop />
+        <VStack gap={16}>
+          <HCenter>
+            <WalletAvatar
+              address={selectedWallet.address}
+              size={104}
+              active={false}
+              borderRadius={24}
+            />
+          </HCenter>
+          <HCenter>
+            {eligibleAddresses.length > 1 && !personalSignMutation.isSuccess ? (
```

Comment:
May be it is better to create a totally different layout for isSuccess style then patch the page in multiple places?

Reply:
Basically both states have the same layout... Not sure if its worth it